### DISCUSSION TITLE
ALab2 with default settings out of Moonray 1.5 hours a frame on 48 cores (AMD EPYC Rome 7402P ) using XPU and 75 Gb texture cache

### DISCUSSION THREAD

**Discussion by samhodge** [dreamworksanimation/openmoonray#98] [Show and tell]
See: 

Animal Logic ALab2:

https://animallogic.com/alab/

https://dpel.aswf.io/alab/

https://user-images.githubusercontent.com/86946/232289446-f6136c86-37ac-4e93-81eb-1c81a2014f7d.mp4

Complete with uncompressed OpenEXR and logfile

[stoatrender.1049.exr.zip](https://github.com/dreamworksanimation/openmoonray/files/11241763/stoatrender.1049.exr.zip)
[stoatrender_log.1049.txt.zip](https://github.com/dreamworksanimation/openmoonray/files/11241764/stoatrender_log.1049.txt.zip)

**Comment by samhodge** [comment]
Feel free to grab the input data as follows

making sure you have `maketx` installed as part of OpenImageIO: 
https://github.com/OpenImageIO/oiio/blob/master/INSTALL.md#installing-from-package-managers

Also assumes `wget`, `unzip`, `awk` installed in a `bash` shell environment, I used Linux CentOS 7.9 and `nvidia-docker`

Allow for about 150 Gb of available disk space and about 100Gb of data downloaded.

```
#!/usr/bin/env bash
USD_PREFIX=https://dpel-assets.aswf.io/usd-alab
USD_ARCHIVE=alab-v2.0.1.zip
TEX_PREFIX=https://aswf-dpel-assets.s3.amazonaws.com/usd-alab
TEX_ARCHIVE=alab-textures.v2.0.1.zip
PROC_ARCHIVE=alab-procedurals.v2.0.1.zip

rm alab-*.zip
wget ${USD_PREFIX}/${USD_ARCHIVE}
wget ${USD_PREFIX}/${PROC_ARCHIVE}
wget ${TEX_PREFIX}/${TEX_ARCHIVE}

unzip ${USD_ARCHIVE}
yes | unzip ${PROC_ARCHIVE} -d ALab
yes | unzip ${TEX_ARCHIVE} -d ALab
rm alab-*.zip
find ./ALab -iname '*.usda' | awk '{print "sed -i.deleteme s/.exr/.tx/g "$0}' | sh
find ./ALab -iname '*.deleteme' | awk '{print "rm -f "$0}' | sh
find ./ALab -iname '*.exr' | awk -F '.exr' '{print "maketx "$0" -o "$1".tx"}' | sh
find ./ALab -iname '*.exr' | awk '{print "rm -f "$0}' | sh
```

Then render as follows, after following the setup guides for the Moonray Hydra Delegate Software.

you need three commands

`hd_usd2rdl`
`moonray`
`denoise`

Which are all part of the Moonray binaries available in the Docker instructions.

See Docker documentation about mounting a path with the above ALab2 Universal Scene Description input data mentioned above.

Note the magic number 75096 is the size of the texture cache used to optimise the amount of texture IO from disk by keeping in a volatile memory cache.


Increasing this number will allow more texture to remain in the cache, to allow for greater CPU occupancy. Decreasing this number will increase render times but have a smaller footprint on machine with smaller amount of system memory.

```
echo -e "SceneVariables {\n   [\"texture_cache_size\"] = 75096,\n}\n" > texture_setting.rdla
cat texture_setting.rdla
for i in {1004..1049}
do
   echo "starting frame ${i}" 
   date
   hd_usd2rdl -in entry.usda -purpose render -time ${i} -camera root/camera01/GEO/renderCam_hrc/renderCam_buffer/renderCam_srt/renderCam -out stoatrender.${i}.rdlb
   moonray -exec_mode xpu -in stoatrender.${i}.rdlb -in texture_setting.rdla -out stoatrender.${i}.exr -info | tee stoatrender_log.${i}.txt
   denoise -mode optix -in stoatrender.${i}.exr -out stoatdenoise.${i}.exr
   echo "ending frame ${i}"
   rm -f stoatrender.${i}.rdlb
   date
done
```

The available DRAM on the machine was 113.2 Gb and the GPUs were 2 x  NVIDIA RTX 3090 24Gb each

**Comment by xXTR0YXx** [comment]
Nice! Even though the rendertimes are incredible high. Also looks like lock_frame_noise is set to "true" because the noise for the background isn't moving. "which is typically undesirable." according to the manuals, but especially when using denoisers I found it better to have the noise pattern locked, otherwise you end up having flickering splotches

**Reply by samhodge** [reply]
I would be interested in seeing a faster render that is as clean in another renderer. I gave a commercial renderer that I have used on feature films 2011-2019 and at the same ray depth with similar BSDF, AA and light samples it was noisy and took similar time. Perhaps you have not tried the scene first hand: 50 lights, fur, about seven thousand odd texture files.
